community/ndp/pom.xml

```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.neo4j</groupId>
+    <artifactId>parent</artifactId>
+    <version>2.3-SNAPSHOT</version>
+    <relativePath>../..</relativePath>
+  </parent>
+
+  <properties>
+    <license-text.header>GPL-3-header.txt</license-text.header>
+    <licensing.prepend.text>notice-gpl-prefix.txt</licensing.prepend.text>
+  </properties>
+
+  <groupId>org.neo4j.build</groupId>
+  <artifactId>neo4j-data-protocol-parent</artifactId>
+  <version>2.3-SNAPSHOT</version>
+
+  <packaging>pom</packaging>
+  <name>Neo4j - Data Protocol Project</name>
+  
+  <licenses>
+    <license>
+      <name>GNU General Public License, Version 3</name>
+      <url>http://www.gnu.org/licenses/gpl-3.0-standalone.html</url>
+      <comments>The software ("Software") developed and owned by Network Engine for
+Objects in Lund AB (referred to in this notice as "Neo Technology") is
+licensed under the GNU GENERAL PUBLIC LICENSE Version 3 to all third
+parties and that license is included below.
+
+However, if you have executed an End User Software License and Services
+Agreement or an OEM Software License and Support Services Agreement, or
+another commercial license agreement with Neo Technology or one of its
+affiliates (each, a "Commercial Agreement"), the terms of the license in
+such Commercial Agreement will supersede the GNU GENERAL PUBLIC LICENSE
+Version 3 and you may use the Software solely pursuant to the terms of
+the relevant Commercial Agreement.
+      </comments>
+    </license>
+  </licenses>
+
+
+  <modules>
```

Comment:
Very good question. We've consciously tried an approach with a fine-grained set of modules. We certainly don't *need* them, we could solve the same things with a single module. 

The intention with the fine-grained modules is that they *force* you to design without adding  circular dependencies between components, vs. doing the same thing with packages where there is nothing stopping someone from introducing bad dependencies across components.

In the other end of the scale are things like the overhead of maintaining this many modules, and of course that it's trickier to deal with if you get the modules themselves wrong in the design.

I've found it to be much less overhead than what I'd anticipated, and must say I quite like the fine-grained modules. I think that, beyond the architecturally-proper straightjacket, it also helps looking at the code since things are in tiny labelled boxes. All that said, I really don't feel strongly about it, I'd be ok swapping it to a single module (or two) if people feel strongly opposed to strict compartmentalization.